lib/main.js

```diff
@@ -394,18 +395,42 @@ const Hydrogen = {
 
   restartKernelAndReEvaluateBubbles() {
     const { editor, kernel } = store;
-    if (!editor || !kernel) {
-      this.runAll();
-      return;
-    }
 
     let breakpoints = [];
     _.forEach(this.markerBubbleMap, (bubble: ResultView) => {
       breakpoints.push(bubble.marker.getBufferRange().start);
     });
     this.clearResultBubbles();
 
-    kernel.restart(() => this.runAll(breakpoints));
+    if (!editor || !kernel) {
+      this.runAll(breakpoints);
+    } else {
+      kernel.restart(() => this.runAll(breakpoints));
+    }
+  },
+
+  toggleBubble() {
+    const { editor } = store;
+    if (!editor) return;
+    const [startRow, endRow] = editor.getLastSelection().getBufferRowRange();
+
+    for (let row = startRow; row <= endRow; row++) {
+      let destroyed = false;
+
+      _.forEach(this.markerBubbleMap, (bubble: ResultView) => {
```

Comment:
I wanted to know the result status(destroyed of not), but it not return result status.

I could write isRowContainsBubbke(row) predicate function to determine add or remove bubble, I just take simple imperative one.